### ISSUE TITLE
Support of endpoint only deployments

### ISSUE THREAD

**Issue by HyperBrain** [serverless/serverless#899]
For endpoints that only return static data (like a fixed descriptor object) I use the `MOCK` endpoint type in my `s-function.json` like this:

```
{
  "name": "descriptor",
  "runtime": "nodejs",
  "description": "fixed decriptor",
  "customName": false,
  "customRole": false,
  "handler": "handler.handler",
  "timeout": 6,
  "memorySize": 128,
  "authorizer": {},
  "custom": {
    "excludePatterns": []
  },
  "endpoints": [
    {
      "path": "descriptor",
      "method": "GET",
      "type": "MOCK",
      "authorizationType": "none",
      "authorizerFunction": false,
      "apiKeyRequired": false,
      "requestParameters": {},
      "requestTemplates": {
        "application/json": "{ statusCode: 200 }"
      },
      "responses": {
        "400": {
          "statusCode": "400"
        },
        "default": {
          "statusCode": "200",
          "responseParameters": {},
          "responseModels": {},
          "responseTemplates": {
            "application/json": "$${descriptorTemplate}"
          }
        }
      }
    }
  ],
  "events": [],
  "environment": {
    "SERVERLESS_PROJECT": "${project}",
    "SERVERLESS_STAGE": "${stage}",
    "SERVERLESS_REGION": "${region}"
  },
  "vpc": {
    "securityGroupIds": [],
    "subnetIds": []
  }
}
```

The fixed result is just retrieved from a template defined in `s-templates.json` with a fixed status code and is completely handled in API Gateway without the need of any associated lambda function.

The current implementation of Serverless only accepts endpoint deploys only if there is an associated function deployed before. For `MOCK` endpoints this strong coupling should be removed as the whole function existence and deployment is unnecessary in that case.

**Comment by eahefnawy** [comment]
We've been discussing taking API gateway config out of `s-function.json` and not have it too tightly coupled with the function for that reason. While your approach would fix the issue, it feels kinda weird to deploy an independent endpoint with no function, yet it is still defined within a function (in `s-function.json`).

Maybe if we're gonna implement this we should just take it out of `s-function.json` somehow? (huge breaking change!!)

**Comment by flomotlik** [comment]
In the future this should be possible through providing your own Cloudformation template or resources so you can have full control over the endpoint but without this special case needing any special code inside of serverless. We should definitely document it though.

**Comment by HyperBrain** [comment]
@flomotlik 
I fixed the dependency already in 0.5.5 -> deploys are currently possible without deploying the function.
Nevertheless :+1: for having no function declaration at all in this case in V1.0

**Comment by flomotlik** [comment]
Ok then I will simply close this issue as it is resolved for now and will be resolved automatically in V1 with our new resource handling.
